hack/release/push-node.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# this script replaces hack/release/build/push-node.sh for Kubernetes v1.31+
+# usage: push-node.sh v1.32.0
+
+set -o errexit -o nounset -o pipefail
+
+REGISTRY="${REGISTRY:-gcr.io/k8s-staging-kind}"
+IMAGE_NAME="${IMAGE_NAME:-node}"
+
+# cd to the repo root
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." &> /dev/null && pwd -P)"
```

Comment:
Can we always assume we will be in a git repo and not running on a downloaded copy of the source? If so, might be slightly safer to use `git rev-parse --show-toplevel`.

Reply:
This approach is safe unless we move the script relative to the repo root, we use it pretty widely in Kubernetes. BASH_SOURCE[0] will be the script invoked and is reliable vs $0 or something like that.

It also works with a downloaded source tarball, for this particular script maybe we don't care, but I prefer using a consistent pattern, and for other scripts some packagers are using tarballs IIRC. (though we currently also have https://github.com/kubernetes-sigs/kind/issues/3645)

This trick is a neat combination of things that are safe versus the more trivial `dirname "${BASH_SOURCE}"`

1) we use a subshell, so we can CD without affecting the current working directory
2) we use CD + `pwd -P` to portably resolve the real path (versus something like realpath/readlink that are portability nightmares)
3) we use BASH_SOURCE[0] to specifically get the primary source script in a supported way instead of depending on something like "$0 will probably maybe have it"

Reply:
(`pwd -P` happens to be specified in posix: https://pubs.opengroup.org/onlinepubs/007904875/utilities/pwd.html and that flag forces it to resolve symlinks, but only for the current working directory)

Reply:
The main downside of this approach is you need to know the levels of nesting to the repo root (`/.../..`) but that only needs changing if we move it to be more or less nested (like this, the other script is `/../../..`), which won't happen often and you're going to catch the first time you run the script after moving it.

Reply:
(er and `BASH_SOURCE` is safe because we specifically use `#!/usr/bin/env bash` and bash specifies this, we usually require bash for development scripts in kubernetes (along with gnu coreutils), I use POSIX sh for some portable utilities that I want end users to run, like the local registry script, but it lacks a lot of modern features and is avoided in favor of pash otherwise)

Reply:
... also I think `git rev-parse --show-toplevel` doesn't work for git worktree, which is something we're starting to use more in Kubernetes scripts (e.g. for diffing codegen in verify scripts)